clients/python-apiserver-client/python_apiserver_client/kuberay_apis.py

```diff
@@ -0,0 +1,261 @@
+            list of clusters
+    """
+    def get_cluster(self, ns: str, name: str) -> tuple[int, str, Cluster]:
+        # Execute HTTP request
+        url = self.base + self.api_base + f"namespaces/{ns}/clusters/{name}"
+        response = requests.get(url, headers=_headers, timeout=(10, 10))
+        # Check execution status
+        if response.status_code // 100 != 2:
+            return response.status_code, response.json()["message"], None
+        return response.status_code, None, cluster_decoder(response.json())
+
+    """
+        create cluster
+        Parameter:
+            cluster definition
+        Returns:
+            http return code
+            message - only returned if http return code is not equal to 200
+    """
+    def create_cluster(self, cluster: Cluster) -> tuple[int, str]:
+        # Execute HTTP request
+        url = self.base + self.api_base + f"namespaces/{cluster.namespace}/clusters"
+        response = requests.post(url, json=cluster.to_dict(), headers=_headers, timeout=(10, 10))
+        if response.status_code // 100 != 2:
+            return response.status_code, response.json()["message"]
+        return response.status_code, None
+
+    """
+        get cluster status
+        Parameter:
+            namespace of the cluster
+            name of the cluster
+        Returns:
+            http return code
+            message - only returned if http return code is not equal to 200
+            cluster status
+    """
+    def get_cluster_status(self, ns: str, name: str) -> tuple[int, str, str]:
+        # Execute HTTP request
+        status, error, cluster = self.get_cluster(ns=ns, name=name)
+        # Check execution status
+        if status // 100 != 2:
+            return status, error, None
+        cluster_status = "creating"
+        if cluster.cluster_status is not None:
+            cluster_status = cluster.cluster_status
+        return status, None, cluster_status
+
+    """
+        wait cluster ready
+        Parameter:
+            namespace of the cluster
+            name of the cluster
+            wait time (-1 waits forever)
+        Returns:
+            http return code
+            message - only returned if http return code is not equal to 200
+            cluster status
+    """
+    def wait_cluster_ready(self, ns: str, name: str, wait: int = -1) -> tuple[int, str]:
+        current_wait = 0
+        while True:
+            status, error, c_status = self.get_cluster_status(ns=ns, name=name)
+            # Check execution status
+            if status // 100 != 2:
+                return status, error
+            if c_status == "ready":
+                return status, None
+            if current_wait > wait > 0:
+                return 408, f"Timed out waiting for cluster ready in {current_wait} sec"
+            time.sleep(self.wait_interval)
+            current_wait += self.wait_interval
+
+    """
```

Comment:
Naive Python question that just occurred to me. I thought all these doc strings must be after the opening line, e.g., `def get_cluster_endpoints(...) -> ...:`?? Isn't that required for documentation tools to extract them correctly? cc: @maxdebayser